src/js/outreach/views/verify_views.js

```diff
@@ -0,0 +1,146 @@
+    this.ui.submit.prop('disabled', true);
+  },
+});
+
+const VerifyCodeView = View.extend({
+  ui: {
+    submit: '.js-submit',
+    resend: '.js-resend',
+    input: '.js-input',
+    errorMessage: '.js-error-message',
+  },
+  triggers: {
+    'click @ui.submit': 'click:submit',
+    'click @ui.resend': 'click:resend',
+  },
+  events: {
+    'input @ui.input': 'watchInput',
+    'keydown @ui.input': 'watchKeydown',
+  },
+  template: hbs`
+    <div class="verify__icon verify__icon--blue">{{fat "user-lock"}}</div>
+    <h2 class="verify__heading-text">Enter your verification code.</h2>
+    <p class="verify__info-text">We sent a text message with a verification code to the phone number XXX-XXX-{{ phoneEnd }}.</p>
+    <div class="verify__code-fields">
+      <input class="input-primary verify__code-input js-input{{#if hasInvalidCodeError}} has-error{{/if}}" />
+      <input class="input-primary verify__code-input js-input{{#if hasInvalidCodeError}} has-error{{/if}}" />
+      <input class="input-primary verify__code-input js-input{{#if hasInvalidCodeError}} has-error{{/if}}" />
+      <input class="input-primary verify__code-input js-input{{#if hasInvalidCodeError}} has-error{{/if}}" />
+    </div>
+    {{#if hasInvalidCodeError}}
+      <p class="verify__error-text js-error-message">Incorrect verification code. Please try again.</p>
+    {{/if}}
+    <button class="verify__submit button--green w-100 js-submit" disabled>Submit</button>
+    <div class="verify__heading-text u-text-link js-resend">Send a new code</div>
+  `,
+  templateContext() {
+    return {
+      phoneEnd: this.getOption('patientPhoneEnd'),
+      hasInvalidCodeError: this.getOption('hasInvalidCodeError'),
+    };
+  },
+  watchInput(event) {
+    if (this.getOption('hasInvalidCodeError')) {
+      this.removeErrorMessaging();
+    }
+
+    const inputElements = this.ui.input;
+    const index = inputElements.index(event.target);
+
+    const value = String(event.target.value);
+    const first = value.charAt(0);
+    const rest = value.substring(1);
+
+    event.target.value = first ?? '';
+
+    const isLastInputEl = index === inputElements.length - 1;
+    const didInsertContent = first !== undefined && value.length;
+
+    const code = inputElements.map((i, el) => el.value).get().join('');
+
+    this.disableSubmitButton(code.length !== inputElements.length);
+
+    if (didInsertContent && !isLastInputEl) {
+      const currentElement = inputElements.eq(index + 1);
+
+      currentElement.focus();
+      currentElement.val(rest);
+      currentElement.trigger('input');
+    }
+  },
+  watchKeydown(event) {
+    const inputElements = this.ui.input;
+    const index = inputElements.index(event.target);
+
+    if (event.keyCode === BACKSPACE_KEY && event.target.value === '') {
+      inputElements.eq(Math.max(0, index - 1)).focus();
+    }
+  },
+  removeErrorMessaging() {
+    this.ui.input.removeClass('has-error');
+    this.ui.errorMessage.remove();
```

Comment:
I don't love altering the DOM outside of the render.  This is also different behavior to other forms.  other forms keep the error state until the next submit.  Check the name input on the clinician sidebar.  Maybe we can bring this in line with those other form errors?

Reply:
Okay, I've updated it so the input error and message stay visible.